### ISSUE TITLE
[P0][M1] case/default labels outside switch silently ignored

### ISSUE THREAD

**Issue by ChaseWNorton** [anthropics/claudes-c-compiler#23]
## Problem

A `case` or `default` label outside of any `switch` statement is silently ignored. C11 §6.8.1p1 requires case and default labels to appear only within a switch body. GCC emits `error: case label not within a switch statement`.

In the IR lowering (`src/ir/lowering/stmt_switch.rs` lines 203-205):

```rust
pub(super) fn lower_case_stmt(&mut self, expr: &Expr, stmt: &Stmt) {
    let case_val = self.eval_case_constant(expr);
    let label = self.fresh_label();
    if let Some(frame) = self.func_mut().switch_stack.last_mut() {
        frame.cases.push((case_val, label));  // silently skipped when stack is empty
    }
```

When `switch_stack` is empty (no enclosing switch), the `if let Some` simply doesn't match, and the case label is silently dropped. Same for `lower_default_stmt` at line 226.

## Expected behavior

```
test.c:2:5: error: 'case' label not within a switch statement
```

## Reproduction

```c
int main(void) {
    case 1:          // should error
        return 1;
    default:         // should error
        return 0;
}
```

Compile with `ccc -S -o /dev/null test.c` — exits 0, no diagnostic.

## Suggested approach

Detect this in **sema** (`src/frontend/sema/analysis.rs`). Add an `in_switch_depth: usize` counter to `SemanticAnalyzer`.

In `analyze_stmt` for `Stmt::Switch`: increment before analyzing body, decrement after.

In `analyze_stmt` for `Stmt::Case` / `Stmt::CaseRange` / `Stmt::Default`:
- If `in_switch_depth == 0`, emit `error: 'case' label not within a switch statement` (or `'default' label`).

This is a simple depth counter — no complex state needed.

## Files to modify

- `src/frontend/sema/analysis.rs` — Add `in_switch_depth: usize` to `SemanticAnalyzer`, increment/decrement in `Stmt::Switch`, check in `Stmt::Case`/`Stmt::Default`

## Tests

```rust
#[test]
fn case_outside_switch() {
    let errors = sema_error_count("int f(void) { case 1: return 1; }");
    assert!(errors > 0);
}

#[test]
fn default_outside_switch() {
    let errors = sema_error_count("int f(void) { default: return 0; }");
    assert!(errors > 0);
}

#[test]
fn case_inside_switch_ok() {
    let errors = sema_error_count("int f(void) { switch(0) { case 1: break; default: break; } return 0; }");
    assert_eq!(errors, 0);
}
```

## Acceptance criteria

- [ ] `case` outside switch produces a compile error
- [ ] `default` outside switch produces a compile error
- [ ] `case`/`default` inside switch still works correctly
- [ ] Clean build: `cargo build --release`
- [ ] All existing tests pass: `cargo test --lib`
- [ ] New tests added for the fix

Part of [MILESTONE] M1: Core Diagnostic Coverage (#40)
